scripts/main/lychee.js

```diff
@@ -1173,3 +1173,80 @@ lychee.getBaseUrl = function () {
 		return location.href.replace(location.hash, "");
 	}
 };
+
+/**
+ * drag album to another one
+ * @param {DragEvent} ev
+ * @returns {void}
+ */
+lychee.startDrag = function (ev) {
+	/** @type string */
+	let type = "album";
+	/** @type ?Element */
+	let div = ev.target.closest("div.album");
+	if (!div) {
+		type = "photo";
+		div = ev.target.closest("div.photo");
+		if (!div) return;
+	}
+	ev.dataTransfer.setData("text", `${type}-${div.dataset.id}`);
+};
+
+/**
+ * drop album
+ * @param {DragEvent} ev
+ * @returns {void}
+ */
+lychee.finishDrag = function (ev) {
+	ev.preventDefault();
+
+	/** @type string */
+	const data = ev.dataTransfer.getData("text");
+	/** @type string */
+	let targetId = ev.target.closest("div.album").dataset.id;
+	if (!targetId || data.substring(6) === targetId) return;
+
+	if (data.startsWith("photo-")) {
+		// photo is dragged
+		contextMenu.photoDrop(data.substring(6), targetId, ev);
+	} else {
+		// album is dragged
+		contextMenu.albumDrop(data.substring(6), targetId, ev);
+	}
+};
+
+/**
+ * Album drag-over callback
+ * @param {DragEvent} ev
+ * @returns {void}
+ */
+lychee.overDrag = function (ev) {
+	ev.preventDefault();
+	/** @type ?Element */
+	let div = ev.target.closest("div.album");
+	if (div) {
+		div.classList.add("album__dragover");
+	}
+};
+
+/**
+ * Album drag-leave callback
+ * @param {DragEvent} ev
+ * @returns {void}
+ */
+lychee.leaveDrag = function (ev) {
+	/** @type ?Element */
+	let div = ev.target.closest("div.album");
```

Comment:
```suggestion
	/** @type ?HTMLDivElement */
	const div = ev.target.closest("div.album");
```

Reply:
For `HTMLDivElement` see https://developer.mozilla.org/en-US/docs/Web/API/HTMLDivElement. I don't know if you know or use MDN, but this is my primary resource. Much better than any questionable tutorial or Stack Overflow.

And if you are more the hardcore guy, the ultimate truth is always: https://html.spec.whatwg.org/multipage/grouping-content.html#the-div-element. Luckily, MDN always links to the primary source, but I find MDN much more reader-friendly.